checker/src/visitors.rs

```diff
@@ -1960,6 +1972,10 @@ impl<'a, 'b: 'a, 'tcx: 'b, E> MirVisitor<'a, 'b, 'tcx, E> {
             .iter()
             .filter(|(p, _)| p.is_rooted_by(&rpath))
         {
+            let elapsed_time_in_seconds = self.start_instant.elapsed().as_secs();
```

Comment:
This is repeated a lot, isn't there somewhere you can put a reached_max_analysis_time? I guess it's cheating to put it into k_limits itself.

Reply:
It is bothering me a bit as well, but not as much as other things. Perhaps I'll come back to this in the future.